internal/storage/localdir.go

```diff
@@ -0,0 +1,117 @@
+package storage
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/nlepage/go-tarfs"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ Storage = &LocalDirectory{}
+
+const DefaultBundleCacheDir = "/var/cache/bundles"
+
+var DefaultLocalDirectory = &LocalDirectory{RootDirectory: DefaultBundleCacheDir}
+
+type LocalDirectory struct {
+	RootDirectory string
+}
+
+func (s *LocalDirectory) Load(_ context.Context, owner client.Object) (fs.FS, error) {
+	bundlePath := filepath.Join(s.RootDirectory, fmt.Sprintf("%s.tgz", owner.GetUID()))
+	bundleFile, err := os.Open(bundlePath)
+	if err != nil {
+		return nil, err
+	}
+	defer bundleFile.Close()
+	tarReader, err := gzip.NewReader(bundleFile)
+	if err != nil {
+		return nil, err
+	}
+	return tarfs.New(tarReader)
+}
+
+func (s *LocalDirectory) Store(_ context.Context, owner client.Object, bundle fs.FS) error {
```

Comment:
should we check to see if the bundle is already stored before attempting to store it again? Or is it because the bundle path is `fmt.Sprintf("%s.tgz", owner.GetUID())` we are always going to load the bundle, regardless if another copy of it is already on-disk? 

I suppose this could be a premature optimization that we could address later.

Reply:
It's actually important that `os.Create` works the way it does (if the file exists, it truncates the file and allows re-writing it).

That way, our reconcile logic is still always just reading the pod logs and writing (or re-writing) what we read from the pod. If the pod logs change, we overwrite what we had stored before.

That gets us back into the territory of what's immutable and what's not, but the behavior in this PR at least matches the existing behavior, which is that if the pod logs change, then the stored bundle changes with it.